app/models/manageiq/providers/ansible_tower/automation_manager/refresher.rb

```diff
@@ -15,18 +16,23 @@ def collect_inventory_for_targets(ems, targets)
 
           _log.info "Collecting inventory for #{target.class} [#{target_name}] id: [#{target.id}]..."
 
-          inventory = ManageIQ::Providers::AnsibleTower::AutomationManager::RefreshWorker::Collector.new(ems)
+          # actually the collector could be build in parse_targeted_inventory
+          collector = ManageIQ::Providers::AnsibleTower::Inventory::Collectors::AutomationManager.new(ems)
 
           _log.info "Collecting inventory...Complete"
-          [target, inventory]
+          [target, collector]
         end
       end
 
-      def parse_targeted_inventory(ems, target, inventory)
+      # FIXME: this can be done in BaseManager
+      def parse_targeted_inventory(ems, target, collector)
         log_header = format_ems_for_logging(ems)
         _log.debug "#{log_header} Parsing inventory..."
         hashes, = Benchmark.realtime_block(:parse_inventory) do
-          ManageIQ::Providers::AnsibleTower::AutomationManager::RefreshWorker::Parser.new(ems, target, inventory).parse
+          inv_target = ManageIQ::Providers::AnsibleTower::Inventory::Targets::AutomationManager.new(target)
+          inv_target.parser = ManageIQ::Providers::AnsibleTower::Inventory::Parsers::AutomationManager.new(collector)
+          inv_target.parse
+          inv_target.inventory_collections
```

Comment:
I'm not crazy about this block ...

I don't think the `inv_target` should contain the `parser`, and I don't think the `inv_target` should delegate to the `parser` to `parse`.

It's more straightforward to me to have:

``` ruby
# target = Active Record Target
# create inventory target
inv_target = ManageIQ::Providers::AnsibleTower::Inventory::Targets::AutomationManager.new(target)
# create inventory parser
parser = ManageIQ::Providers::AnsibleTower::Inventory::Parsers::AutomationManager.new(collector)
# parser parses target using given inventory collector
parser.parse(inv_target)
# parser returns parsed inventory collections
parser.inventory_collections
```

Or, even have the `parse` method return the `inventory_collections`?

But, I can't wrap my head about having the `inv_target` knowing how to parse itself.  Happy to talk about it, maybe I just need to see it from a different perspective.